google-cloud-spanner/src/test/java/com/google/cloud/spanner/connection/StatementParserTest.java

```diff
@@ -1266,6 +1266,178 @@ public void testPostgreSQLGetQueryParameters() {
+    assertTrue(
+        parser
+            .parse(Statement.of("insert into x (a,b) values (1,2) then return(a)"))
+            .hasReturningClause());
+    assertTrue(
+        parser
+            .parse(Statement.of("insert into x (a,b) values (1,2)then return(a)"))
+            .hasReturningClause());
+    assertTrue(
+        parser
+            .parse(Statement.of("insert into x (a,b) values (1,2) then/*comment*/return *"))
+            .hasReturningClause());
+    assertTrue(
+        parser
+            .parse(Statement.of("insert into x (a,b) values (1,2) then return /*then return*/ *"))
+            .hasReturningClause());
+    assertTrue(
+        parser
+            .parse(Statement.of("insert into x (a,b) values (1,2)then/*comment*/return *"))
+            .hasReturningClause());
+    assertTrue(
+        parser
+            .parse(Statement.of("insert into x (a,b) values (1,2)then/*comment*/return(a)"))
+            .hasReturningClause());
+    assertTrue(
+        parser
+            .parse(Statement.of("insert into x (a,b) values (1,2)then /*comment*/return(a)"))
+            .hasReturningClause());
+    assertTrue(
+        parser
+            .parse(
+                Statement.of("insert into x (a,b) values (1,2)/*comment*/then/*comment*/return(a)"))
+            .hasReturningClause());
+    assertTrue(
+        parser
+            .parse(
+                Statement.of(
+                    "insert into x (a,b) values (1,2)/*comment*/then/*comment*/return/*comment*/(a)"))
+            .hasReturningClause());
+    assertTrue(
+        parser
+            .parse(
+                Statement.of(
+                    "insert into x (a,b) values (1,2)/*comment"
+                        + "*/then"
+                        + "/*comment"
+                        + "*/return/*"
+                        + "comment*/(a)"))
+            .hasReturningClause());
+    assertTrue(
+        parser
+            .parse(Statement.of("delete from x where y=\"z\"then return *"))
+            .hasReturningClause());
+    assertTrue(
+        parser
+            .parse(Statement.of("insert into x select 'then return' as returning then return *"))
+            .hasReturningClause());
+    assertTrue(
+        parser.parse(Statement.of("delete from x where 10=`z`then return *")).hasReturningClause());
+    assertFalse(
+        parser
+            .parse(Statement.of("insert into x (a,b) values (1,2) returning (a)"))
+            .hasReturningClause());
+    assertFalse(
+        parser
+            .parse(Statement.of("insert into x (a,b) values (1,2) /*then return **/"))
+            .hasReturningClause());
+    assertFalse(
+        parser.parse(Statement.of("insert into x (a,b) values (1,2)")).hasReturningClause());
+    assertFalse(
+        parser
+            .parse(Statement.of("insert into x (a,b) values (1,2)thenreturn*"))
+            .hasReturningClause());
+  }
+
+  @Test
+  public void testPostgreSQLReturningClause() {
```

Comment:
PostgreSQL supports some additional syntaxes for strings, [Dollar-Quoted String Constants](https://www.postgresql.org/docs/current/sql-syntax-lexical.html#SQL-SYNTAX-DOLLAR-QUOTING) in particular, that you may want to add test cases for. For instance,

```sql
UPDATE foo SET bar = $$RETURNING a, b, c$$ WHERE baz = 123
```

does not have a returning clause. Nor does,


```sql
UPDATE foo SET bar = $foobar$RETURNING a, b, c$foobar$ WHERE baz = 123
```

nor,

```sql
UPDATE foo SET bar = $returning$Hello, world!$returning$ WHERE baz = 123
```

Reply:
@c2nes: Thanks for the test cases! Added these as well in https://github.com/googleapis/java-spanner/pull/1978/commits/9f59715e76511dceb43825a66c06f290978ed02e.